stdlib/public/core/Filter.swift

```diff
@@ -203,8 +205,28 @@ public struct LazyFilterCollection<
 
   // TODO: swift-3-indexing-model - add docs
   @warn_unused_result
-  public func next(index: Index) -> Index {
-    return LazyFilterIndex(base: _nextFiltered(index.base))
+  public func next(i: Index) -> Index {
+    return LazyFilterIndex(base: _nextFiltered(i.base))
+  }
+
+  // TODO: swift-3-indexing-model - add docs
+  @warn_unused_result
+  public func advance(i: Index, by n: IndexDistance) -> Index {
+    var index = i.base
+    for _ in 0..<n where index != _base.endIndex {
+      index = _nextFiltered(index)
+    }
+    return LazyFilterIndex(base: index)
+  }
+
+  // TODO: swift-3-indexing-model - add docs
+  @warn_unused_result
+  public func advance(i: Index, by n: IndexDistance, limit: Index) -> Index {
+    var index = i.base
+    for _ in 0..<n where index != _base.endIndex && index != limit.base {
```

Comment:
Yeah I was experimenting with this form and planned to look at the code emitted to see if the loop could be optimized to stop once one of the where condition got hit. ...then I forgot to actually do that. I will change it to other form.